pkg/apimachinery/patch/patch.go

```diff
@@ -38,9 +37,73 @@ const (
 	PatchRemoveOp  = "remove"
 )
 
+type PatchSet interface {
+	Test(path string, value interface{})
+	Add(path string, value interface{})
+	Remove(path string)
+	Replace(path string, value interface{})
+	TestAndReplace(path string, oldVal, newVal interface{})
+	AddOrReplace(path string, value interface{}, condition bool)
+	IsEmpty() bool
+	GeneratePayload() ([]byte, error)
+}
+
+type patchSet struct {
+	patches []PatchOperation
+}
+
+func New() PatchSet {
```

Comment:
It should return a concrete type, not an interface. The caller can use it through an interface if it wants to.

Reply:
Ok, I can make it a type not an interface